### ISSUE TITLE
Allow Spectator to see traps/assassins

### ISSUE THREAD

**Issue by GRhin** [Sh0wdown/UnofficialCrusaderPatch#408]
Is it possible to allow the player, when in spectator mode, to see all players traps and assassins?
E.G be able to see where the caliph has laid his pitch.
I figure if there were a way it would be through a true/false value somewhere of being any ally. Depending on how the game treats teams, if its a true/false of being on a certain team, then we might be able to set the spectator as part of all teams, if its a team = # type thing, then it wont work like that.

**Comment by Admiral-Gumpert** [comment]
Would making the spectator an ally to all opponents in the game affect the battle? like how to win or how to kill the remaining player "spectator" , Idk I like your idea by the way

**Comment by GRhin** [comment]
I dont think so, afaik the "spectator" is similar to a dead player continuing to watch in an online game. If the game thinks the spectator is dead, their being on any or all teams shouldnt effect win conditions. Of course this is all dependant on how the code actually works, so I may be way off base, but at least its a start point for someone with the knowledge to actually try get this in place, if its that simple then this would be a very quick tweak to put in place, assuming someone knows where to look for those particular settings, if its not that simple, it could be a lot of work.

**Comment by Krarilotus** [comment]
Actually you can do that alrdy. I made a map, the AI Contest map, where you can spectate while beeing able to build yourself / control yourself. But be carefull, as soon as you touch Troops of the AI, he ll not control them again. But you can even build farms like the AI over trees and stuff like that. Your AI will still function like normal and you can look in his book.

I still don t know exactly how i got it working, but i ll figure it out. Then i can post an instruction :)

**Comment by Rakso69** [comment]
@Krarilotus Upload map please

**Comment by Krarilotus** [comment]
> @Krarilotus Upload map please

I will upload the map on the weekend. Because it will be part of my AI contest, i dont want anybody to have an unfair advantage by obtaining the map prior to the announcement.

I will send you the link via youtubecomment or if you give me an email or a way to contact you directly i will give you the link as soon as it goes up.

Keep up the AI advancements!

**Comment by Rakso69** [comment]
@Krarilotus I don't want to take part in. I won't finish my AI in time for saturday, don't consider me a partecipant. Feel free to give me the map. I private messaged you on Discord.

**Comment by GRhin** [comment]
@Krarilotus I kinda meant the ability to see all players traps/assassins, for example seeing when The Sherrif's assassins crossing The Caliph's pitch ditches.
Also, I dont know if this is there or not as I havent looked, but what happens if you spectate a 3+ player free for all game, and the assasins of player 2 is killing player 3, I assume that player 1 (and the spectator) cannot see the assasins but can see the damage they are doing? I think player 1 should NOT be able to see them, but itd be nice for the spectator to be able to see them, the tweak I am asking for should also resolve that situation.

**Comment by GRhin** [comment]
So I played Krarilotus's map last night with a barebones AI and as spectator i could control almost the entire player 1 account (except for some units i couldnt move), but every time i killed a lord using the player 1 account, the wolf (who was used for the barebones AI) would send me a message congratulating me. So theres a little bit of "ally" there but also a bit of "there are two player 1's". Im not sure if that makes this suggestion unworkable or not.

**Comment by GRhin** [comment]
> Also, I dont know if this is there or not as I havent looked, but what happens if you spectate a 3+ player free for all game, and the assasins of player 2 is killing player 3, I assume that player 1 (and the spectator) cannot see the assasins but can see the damage they are doing? I think player 1 should NOT be able to see them, but itd be nice for the spectator to be able to see them, the tweak I am asking for should also resolve that situation.

I had this situation recently without spectator, basically assassins that can be seen by one player (aside from their owner) can be seen by all players. In the example above, Player 2 can see player 3s assassins attacking him, so Player 1 can also see them, despite not being allied to either player. I dont think this is 'logical' but I dont think its too big of a problem, and it means that the spectator could also see them in that case too, It would be nice to always see them in the future, but its bearable as is for the timebeing.

**Comment by Sh0wdown** [comment]
In the next version Assassins will be shown to the spectator. Not sure if I can do the same for traps though.

**Comment by GRhin** [comment]
@sh0wdown you are an absolute legend!

Out of curiosity how come both aren't part of the same setting? I would have assumed it was an all or nothing thing.

**Comment by Krarilotus** [comment]
@GRhin traps are not visible ever, while assassins have a range. So i guess he just makes this range bigger while with the traps he has to find the invisibility property, by switching from  the active to an inactive player, which might be way harder

**Comment by GRhin** [comment]
If that is the case, this feature would not be practical for use in multiplayer, just in offline AI-only matches, which is perfectly fine as currently spectator doesn't.work in multiplayer anyways.
